### ISSUE TITLE
adopt dbt-postgres's method of 

### ISSUE THREAD

**Issue by swanderz** [dbt-msft/dbt-sqlserver#163]
copy the implementation introduced for dbt-postgres in https://github.com/dbt-labs/dbt/pull/3106/

**Comment by davidclarance** [comment]
@swanderz this implementation is neat! could I work on this issue or is someone already working on it?

**Comment by semcha** [comment]
Hi @swanderz!
Do I understand correctly that we need to implement this macro for SQL Server and name it `sqlserver__get_create_index_sql`?
https://github.com/dbt-labs/dbt-core/pull/3106/files#diff-49fbdf0ee41b3d9ec139a881dfed1e9196c47ba014f946421f636bbf57703a6c

**Comment by swanderz** [comment]
> Hi @swanderz! Do I understand correctly that we need to implement this macro for SQL Server and name it `sqlserver__get_create_index_sql`? https://github.com/dbt-labs/dbt-core/pull/3106/files#diff-49fbdf0ee41b3d9ec139a881dfed1e9196c47ba014f946421f636bbf57703a6c

yep! additionally we're need a `SQLServerIndexConfig` for the index definition json. For another example check out [dbt-firebolt's FireboltIndexConfig](https://github.com/firebolt-db/dbt-firebolt/blob/4f97ca73b0585ba8b0103c2838ad25f37c2ad7cb/dbt/adapters/firebolt/impl.py#L19-L85)

**Comment by semcha** [comment]
Working on that issue here: https://github.com/semcha/dbt-sqlserver/commits/new-indexes

**Comment by joshmarkovic** [comment]
Does #714 close this?
